File: 82. Remove Duplicates from Sorted List II/82. Remove Duplicates from Sorted List II.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
    let dummyNode =new ListNode(0,head)
     
    let firstPtr = dummyNode
    let secondPtr = head
    
    while( (secondPtr !== null) &&  secondPtr.next!==null){
        if(secondPtr.val == secondPtr.next.val){
            while( (secondPtr.next !== null) && ( secondPtr.val ==secondPtr.next.val)){
                secondPtr = secondPtr.next
            }
            firstPtr.next=secondPtr.next
            secondPtr=secondPtr.next
        }else{
            firstPtr=firstPtr.next
            secondPtr = secondPtr.next
        }
    }
    
    return dummyNode.next
};

File: 82. Remove Duplicates from Sorted List II/82. Remove Duplicates from Sorted List II.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:

        dummy = ListNode()
        left_ptr = dummy
        right_ptr = head

        while(right_ptr and right_ptr.next):
            if(right_ptr.val != right_ptr.next.val):
                left_ptr.next=right_ptr
                left_ptr=left_ptr.next
                right_ptr = right_ptr.next
            else:
                temp = right_ptr.val
                while(right_ptr  and right_ptr.val == temp):
                    right_ptr=right_ptr.next

        left_ptr.next=right_ptr

        return dummy.next
